### PR TITLE
Pt 154885468 custom http api router

### PR DESCRIPTION
**Pull request by fabiankrol**

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -44,7 +44,7 @@
         {idna, {git, "https://github.com/aeternity/erlang-idna", {ref, "24bf647"}}}
        ]}.
 
-{plugins, [{swagger_endpoints, {git, "https://github.com/aeternity/swagger_endpoints", {tag, "v0.1.0"}}}]}.
+{plugins, [{swagger_endpoints, {git, "https://github.com/aeternity/swagger_endpoints", {branch, "v0.2.0-wip"}}}]}.
```
Comment (fabiankrol) [inline rebar.config]: TODO update to tag after merging: https://github.com/ThomasArts/swagger_endpoints/pull/3

File: apps/aehttp/src/aehttp_api_router.erl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    Handler :: aehttp_api_handler,
+    InitArgs :: init_args()
+}].
+
+get_paths(Target, LogicHandler) ->
+    % Swagger validation will be replaced
+    % we're using it as a intermediate step
+    [{'_', [
+            {_, _, {_, _, SwaggerValidator}}
+            | _ ]}] = swagger_router:get_paths(LogicHandler),
```
Comment (fabiankrol) [inline apps/aehttp/src/aehttp_api_router.erl]: Dirty workaround 🙊

File: apps/aehttp/src/aehttp_api_router.erl [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    Path1 = binary:replace(Path0, <<"}">>, <<"">>, [global]),
+    binary:replace(Path1, <<"{">>, <<":">>, [global]).
+
+method(Atom) ->
+    list_to_binary(string:uppercase(atom_to_list(Atom))).
+
+is_enabled(Target, Tags) when is_atom(Target) ->
+    is_enabled(atom_to_binary(Target, utf8), Tags);
+is_enabled(Target, Tags) ->
+    lists:member(Target, Tags).
```
Comment (fabiankrol) [inline apps/aehttp/src/aehttp_api_router.erl]: @velzevur ✨

File: apps/aeutils/src/aeu_http_client.erl [hunk 4]
```diff
@@ -59,6 +61,12 @@ process_http_return(R) ->
 encode_get_params(#{} = Ps) ->
     encode_get_params(maps:to_list(Ps));
 encode_get_params(Params) when is_list(Params) ->
-    ["?", lists:join( "&", [ [http_uri:encode(K), "=" , http_uri:encode(V)] 
+    ["?", lists:join( "&", [ [http_uri:encode(to_string(K)), "=" , http_uri:encode(to_string(V))]
```
Comment (ThomasArts) [inline apps/aeutils/src/aeu_http_client.erl]: good: I will later remove places where caller converts to string, such as:
https://github.com/aeternity/epoch/blob/5dc06ae79f61ca897ffc5a620a09dc7d5404f21d/apps/aeutils/src/aeu_requests.erl#L149

Comment (fabiankrol) [reply apps/aeutils/src/aeu_http_client.erl]: 👍

**Pull request merged**